File: JavaApp/src/main/java/com/charlezz/javaapp/feature/local/UserDao.java
package com.charlezz.javaapp.feature.local;

import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public interface UserDao {
    @Query("SELECT * FROM User")
    DataSource.Factory<Integer, User> getUsers();

    @Insert
    void insertUser(User user);
}

File: app/src/main/java/com/charlezz/android/db/UserDao.kt
package com.charlezz.android.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.charlezz.android.db.entity.User

@Dao
interface UserDao {

    @Query("SELECT * FROM user where id==:id")
    suspend fun getUser(id: Int):User?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(posts: List<User>)

    @Query("DELETE FROM user")
    fun deleteAll()
}